### PR TITLE
Jlc/mfe config api nelp

### PR DESCRIPTION
**Pull request by johanv26**
# Description

Add eox-nelp flavour of mfe config api.
this add to the MFE-CONFIG response two changes : 
- all the `THEMING_OPTIONS` object 
-  the `interactive_color` nested in the form: CUSTOM_PRIMARY_COLOR': {'primary': interactive_color},`

## Before 

image

## After
image

### gif comparison
Peek 2023-02-06 15-02

# BreakChanges:

- This removes the studio entry for eox-nelp due lms config API is activated  only in the lms.
- from now the package also depends on eox-theming.

### CODE REVIEW

File: eox_nelp/views.py [hunk 1]
```diff
@@ -34,3 +40,26 @@ def info_view(request):
...
     )
+
+
+class NelpMFEConfigView(MFEConfigView):
+    """
+    Provides an API endpoint to get the MFE_CONFIG from site configuration with
+    nelp extra fields.
+    """
+
+    @method_decorator(cache_page(settings.MFE_CONFIG_API_CACHE_TIMEOUT))
```
Comment (andrey-canon) [inline eox_nelp/views.py:51]: is this necessary, since you are calling super ?

File: eox_nelp/views.py [hunk 2]
```diff
@@ -34,3 +40,26 @@ def info_view(request):
...
+    def get(self, request):
+        """
+        Return the MFE configuration by NELP(adding custom nelp fields).
+        """
+        base_get_response = super().get(request)
```
Comment (andrey-canon) [inline eox_nelp/views.py:56]: all this logic is based on a status 200, so please verify that  the super method returns that status otherwise  just return the same response

File: eox_nelp/views.py [hunk 3]
```diff
@@ -34,3 +40,26 @@ def info_view(request):
...
+        mfe_config_dict = json.loads(base_get_response.content)
+        theme_options = theming.options('THEME_OPTIONS')
+        interactive_color = theming.options('interactive_color')
+        theme_additions = {
+            'THEME_OPTIONS': theme_options,
+            'CUSTOM_PRIMARY_COLOR': {'primary': interactive_color},
```
Comment (andrey-canon) [inline eox_nelp/views.py:62]: it almost official, that this should be  `png-color-primary-base`

File: setup.py [hunk 4]
```diff
@@ -94,8 +94,5 @@ def get_version(*file_paths):
         "lms.djangoapp": [
             'eox_nelp = eox_nelp.apps:EoxNelpConfig',
         ],
-        "cms.djangoapp": [
```
Comment (andrey-canon) [inline setup.py:97]: why did you remove this ?

File: eox_nelp/views.py [hunk 5]
```diff
@@ -6,11 +6,17 @@
...
 import six
-from django.http import HttpResponse
 
+from django.conf import settings
+from django.http import HttpResponse, JsonResponse
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import cache_page
+from rest_framework import status
+
+from lms.djangoapps.mfe_config_api.views import MFEConfigView
```
Comment (andrey-canon) [inline eox_nelp/views.py:17]: this should be a new backend

File: setup.py (hunk 4 above)
Comment (johanv26) [reply setup.py:97]: Studio doesn't have mfe config app  and eox-theming configurations.



So this plugin should not work in studio. or what another approach could we have? CHanging the entry URLs?

File: eox_nelp/views.py (hunk 1 above)
Comment (johanv26) [reply eox_nelp/views.py]: [2399c52](https://github.com/eduNEXT/eox-nelp/pull/17/commits/2399c52d60eebf2823e494f685b8be276f19da20)

File: eox_nelp/views.py (hunk 2 above)
Comment (johanv26) [reply eox_nelp/views.py:53]: [7444ad5](https://github.com/eduNEXT/eox-nelp/pull/17/commits/7444ad5282b135b8f1397c78415911a52e99a24e)

File: eox_nelp/views.py (hunk 3 above)
Comment (johanv26) [reply eox_nelp/views.py]: [cd92192](https://github.com/eduNEXT/eox-nelp/pull/17/commits/cd92192612f88edeed0a34be248fd28c4dec7ca6)

File: eox_nelp/views.py (hunk 5 above)
Comment (johanv26) [reply eox_nelp/views.py]: [2eae781](https://github.com/eduNEXT/eox-nelp/pull/17/commits/2eae78173dbf22725459db90c93f7adbe44f5168)

File: eox_nelp/edxapp_wrapper/mfe_config_view.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+"""
+Wrapper mfe_config api view.
+
+This contains  the required api view from mfe_config
+
+Attributes:
+    backend:Imported site_configuration module by using the plugin settings.
+    MFEConfigView: Wrapper mfe_config  api view class.
+
```
Comment (andrey-canon) [inline eox_nelp/edxapp_wrapper/mfe_config_view.py:9]:
```suggestion
```

File: eox_nelp/views.py [hunk 7]
```diff
@@ -34,3 +37,27 @@ def info_view(request):
...
+    """
+
+    def get(self, request):
+        """
+        Return the MFE configuration by NELP(adding custom nelp fields).
+        """
+        base_get_response = super().get(request)
+        if base_get_response.status_code != 200:
+            return base_get_response
+        mfe_config_dict = json.loads(base_get_response.content)
```
Comment (andrey-canon) [inline eox_nelp/views.py:55]:
```suggestion
        base_get_response = super().get(request)

        if base_get_response.status_code != 200:
            return base_get_response

        mfe_config_dict = json.loads(base_get_response.content)
```

File: eox_nelp/views.py [hunk 8]
```diff
@@ -34,3 +37,27 @@ def info_view(request):
...
+        theme_options = theming.options('THEME_OPTIONS')
+        interactive_color = theming.options('interactive_color')
+        theme_additions = {
+            'THEME_OPTIONS': theme_options,
+            'pgn-color-primary-base': interactive_color,
+        }
+        mfe_config_dict.update(theme_additions)
+        return JsonResponse(mfe_config_dict, status=status.HTTP_200_OK)
```
Comment (andrey-canon) [inline eox_nelp/views.py:63]:
```suggestion
        mfe_config_dict.update(theme_additions)

        return JsonResponse(mfe_config_dict, status=status.HTTP_200_OK)
```

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: setup.py (hunk 4 above)
Comment (johanv26) [reply setup.py]: solved in [18130d3](https://github.com/eduNEXT/eox-nelp/pull/17/commits/18130d3eb0b784ac3ea29e52a45d20f401d7ac64) and [e2cd6d5](https://github.com/eduNEXT/eox-nelp/pull/17/commits/e2cd6d5ce0d3d1a6d8169f874fcaeb2f8a2a28b5)

**Review by andrey-canon — APPROVED** [review]

**Comment by andrey-canon** [comment]
please squash commits
